src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -28,61 +28,83 @@ import { isKeplrHDWallet } from 'lib/utils'
 import { selectIsSnapshotApiQueriesPending, selectVotingPower } from 'state/apis/snapshot/selectors'
 import {
   selectHasUserEnteredAmount,
+  selectInputBuyAsset,
   selectInputSellAsset,
   selectIsAnyAccountMetadataLoadedForChainId,
 } from 'state/slices/selectors'
+import { tradeInput } from 'state/slices/tradeInputSlice/tradeInputSlice'
 import {
   selectActiveQuote,
+  selectBuyAmountAfterFeesCryptoPrecision,
+  selectBuyAmountAfterFeesUserCurrency,
   selectFirstHop,
   selectIsTradeQuoteRequestAborted,
   selectIsUnsafeActiveQuote,
   selectShouldShowTradeQuoteOrAwaitInput,
 } from 'state/slices/tradeQuoteSlice/selectors'
 import { tradeQuoteSlice } from 'state/slices/tradeQuoteSlice/tradeQuoteSlice'
 import { useAppDispatch, useAppSelector } from 'state/store'
-import { breakpoints } from 'theme/theme'
 
 import { useAccountIds } from '../../hooks/useAccountIds'
+import { SharedTradeInput } from '../SharedTradeInput/SharedTradeInput'
 import { CollapsibleQuoteList } from './components/CollapsibleQuoteList'
-import { ConfirmSummary } from './components/ConfirmSummary'
-import { TradeInputBody } from './components/TradeInputBody'
-import { TradeInputHeader } from './components/TradeInputHeader'
-import { WithLazyMount } from './components/WithLazyMount'
-import { useSharedHeight } from './hooks/useSharedHeight'
+import { TradeSettingsMenu } from './components/TradeSettingsMenu'
 
 const votingPowerParams: { feeModel: ParameterModel } = { feeModel: 'SWAPPER' }
+const acknowledgementBoxProps = {
+  display: 'flex',
+  justifyContent: 'center',
+}
 
 const STREAM_ACKNOWLEDGEMENT_MINIMUM_TIME_THRESHOLD = 1_000 * 60 * 5
 
 type TradeInputProps = {
   tradeInputRef: React.MutableRefObject<HTMLDivElement | null>
   isCompact?: boolean
+  onChangeTab: (newTab: TradeInputTab) => void
 }
 
-export const TradeInput = ({ isCompact, tradeInputRef }: TradeInputProps) => {
+export const TradeInput = ({ isCompact, tradeInputRef, onChangeTab }: TradeInputProps) => {
   const {
     dispatch: walletDispatch,
     state: { isConnected, isDemoWallet, wallet },
   } = useWallet()
-  const bodyRef = useRef<HTMLDivElement | null>(null)
-  const totalHeight = useSharedHeight(tradeInputRef)
-  const [isSmallerThanXl] = useMediaQuery(`(max-width: ${breakpoints.xl})`, { ssr: false })
+
   const { handleSubmit } = useFormContext()
   const dispatch = useAppDispatch()
+  const translate = useTranslate()
   const mixpanel = getMixPanel()
   const history = useHistory()
   const { showErrorToast } = useErrorHandler()
+  const { manualReceiveAddress, walletReceiveAddress } = useReceiveAddress({
+    fetchUnchainedAddress: Boolean(wallet && isLedger(wallet)),
+  })
+  const {
+    sellAssetAccountId: initialSellAssetAccountId,
+    buyAssetAccountId: initialBuyAssetAccountId,
```

Comment:
See comment above re: do we want to stop destructure-renaming this, or perhaps use `selected` terminology instead?